launch_ros/launch_ros/actions/node.py

```diff
@@ -365,7 +367,24 @@ def execute(self, context: LaunchContext) -> Optional[List[Action]]:
                     '{}:={}'.format(remapping_from, remapping_to)
                 )
         context.extend_locals({'ros_specific_arguments': ros_specific_arguments})
-        return super().execute(context)
+        ret = super().execute(context)
+
+        execute_process_logger = launch.logging.get_logger(self.name)
+        if '<node_name_unspecified>' in self.node_name:
+            add_node_name(context, None)
+            execute_process_logger.warning(
```

Comment:
I wouldn't show a warning. It's almost the same that making the `node_name` a requirement (a little more permissive), and I think that should be discussed further.

Reply:
If we want to somewhat add a checking when `node_name` is not specified, we could keep track of the executables path and check if they were twice executed without passing a node name. That will not detect collisions with node names explicitly passed in a launch file.

Reply:
I am unsure how effectively we can make use of the `node_executable` argument in place of `node_name`. I saw some example launch files even under the `ros2` org where all the `Node` instances' `node_executable` arguments were just `python3`, and the difference was the Python script specified in the `argument` argument. If we have to track the uniqueness of the `node_executable` + `argument` combo, that adds a lot of complexity.